sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -379,22 +374,35 @@ private[hive] class HiveClientImpl(
           s"Hive ${version.fullVersion} does not support altering database location")
       }
     }
-    client.alterDatabase(
+    val hiveDb = toHiveDatabase(database, false)
+    client.alterDatabase(database.name, hiveDb)
+  }
+
+  private def toHiveDatabase(database: CatalogDatabase, isCreate: Boolean): HiveDatabase = {
+    val props = database.properties
+    val dbOwner = props.getOrElse(PROP_OWNER_NAME, if (isCreate) userName else null)
+    val dbOwnerType = props.getOrElse(PROP_OWNER_TYPE, "USER")
```

Comment:
if `isCreate = false`, shall we use null or `USER`?

Reply:
BTW is null the same as ""?

Reply:
for `create db` we may forbid this and always use `userName` and `PrincipalType.USER`

Reply:
I have tested with hive before, we accidentally erase `ownerName` and `ownerType` both, but the metastore return owner with null and displayed "" in hive, the owner type is displayed `USER` even it has been erased.